src/engine/translate.ts

```diff
@@ -1,49 +1,37 @@
-import { AudienceDefinition, EngineCondition } from '../../types';
-import { isStringArray, isVectorQueryValue } from '../utils';
+import {
+  AudienceDefinition,
+  EngineCondition,
+  EngineConditionQuery,
+  AudienceDefinitionFilter,
+} from '../../types';
 
+/*
+ * I'm maintaining the union over the translation layer
+ * so the compiler can discriminate it further bellow in the computation
+ * TODO Better audienceDefinition validation
```

Comment:
What do you have in mind?

Reply:
This transformation is actually an identity over its attributes. Being so, additional validation here is likely not needed given the params were validated in the above layers.

Actually, this function is not exactly a translation, or an adapter. This is a union between the audience definition attributes and the rule attributes. The rule ones are just constants for the moment, but I assume they will not remain this way much longer, right?